src/client/content_mapblock.cpp

```diff
@@ -422,7 +422,7 @@ void MapblockMeshGenerator::drawSolidNode()
 			if (cur_node.f->drawtype == NDT_LIQUID) {
 				if (cur_node.f->sameLiquidRender(f2))
 					continue;
-				backface_culling = f2.solidness || f2.visual_solidness;
+				backface_culling = f2.solidness || f2.visual_solidness || f2.drawtype == NDT_NODEBOX;
```

Comment:
A better solution would be to assign `visual_solidness = 1` in `ContentFeatures::updateTextures`. There you'll see that `NDT_MESH` might suffer from the same issue.